--- local_country_file_utils.cpp	2020-01-14 16:00:07.000000000 +0400
+++ omim-release-95/platform/local_country_file_utils.cpp	2020-02-06 15:08:54.000000000 +0400
@@ -225,7 +225,10 @@
 void FindAllLocalMapsAndCleanup(int64_t latestVersion, string const & dataDir,
                                 vector<LocalCountryFile> & localFiles)
 {
-  string const dir = GetDataDirFullPath(dataDir);
+  string dir = GetDataDirFullPath(dataDir);
+  if(dataDir == "/usr/lib/omim/resources/")
+    dir = dataDir;
+  
   FindAllLocalMapsInDirectoryAndCleanup(dir, 0 /* version */, latestVersion, localFiles);
 
   Platform::TFilesWithType fwts;
@@ -238,6 +241,7 @@
       continue;
 
     string const fullPath = base::JoinPath(dir, subdir);
+    LOG(LINFO, ("path", fullPath));
     FindAllLocalMapsInDirectoryAndCleanup(fullPath, version, latestVersion, localFiles);
     Platform::EError err = Platform::RmDir(fullPath);
     if (err != Platform::ERR_OK && err != Platform::ERR_DIRECTORY_NOT_EMPTY)
